collie/interactions/datasets.py

```diff
@@ -342,62 +347,63 @@ def __init__(self,
         # reach a limit of ``max_number_of_samples_to_consider`` tries
         if self.check_num_negative_samples_is_valid:
             print('Checking ``num_negative_samples`` is valid...')
             if self.negative_sample_type == 'item':
                 if self._checked_num_negative_item_samples_is_valid is False:
-                    self._checked_num_negative_item_samples_is_valid = (
-                        self._check_num_negative_samples_is_valid()
-                    )
+                    self._check_num_negative_samples_is_valid(self.negative_sample_type)
+                    self._checked_num_negative_item_samples_is_valid = True
             elif self.negative_sample_type == 'user':
                 if self._checked_num_negative_user_samples_is_valid is False:
-                    self._checked_num_negative_user_samples_is_valid = (
-                        self._check_num_negative_samples_is_valid()
-                    )
+                    self._check_num_negative_samples_is_valid(self.negative_sample_type)
+                    self._checked_num_negative_user_samples_is_valid = True
 
-        self.positive_sets = {}
+        self.positive_set = {}
         if self.max_number_of_samples_to_consider > 0:
             print(f'Generating positive {self.negative_sample_type}s set...')
             self._generate_positive_set()
 
-    def _check_num_negative_samples_is_valid(self) -> bool:
+    def _check_num_negative_samples_is_valid(
+        self,
+        negative_sample_type: Literal['user', 'item']
+    ) -> bool:
         """Check if the number of negative samples is valid."""
-        if self.negative_sample_type == 'item':
+        if negative_sample_type == 'item':
             counter = collections.Counter(self.mat.row)
             num_ids = self.num_items
             opposite_type = 'user'
-        elif self.negative_sample_type == 'user':
+        elif negative_sample_type == 'user':
             counter = collections.Counter(self.mat.col)
             num_ids = self.num_users
             opposite_type = 'item'
 
         max_number_interacted_with = counter.most_common(1)[0][1]
         print(
-            f'Maximum number of {self.negative_sample_type}s a {opposite_type} has interacted with:'
+            f'Maximum number of {negative_sample_type}s a {opposite_type} has interacted with:'
             f' {max_number_interacted_with}'
         )
 
         del counter
 
-        is_valid = (
-            self.num_negative_samples
-            < (num_ids - max_number_interacted_with)
-        )
+        num_ids_max_interactions_diff = num_ids - max_number_interacted_with
+
+        is_valid = self.num_negative_samples < num_ids_max_interactions_diff
+
         assert is_valid, '``num_negative_samples`` must be less than {}!'.format(
-            (num_ids - max_number_interacted_with)
+            (num_ids_max_interactions_diff)
```

Comment:
I think we can make this an f-string too!

Reply:
I actually noticed there are a lot of `.format()` strings instead of f strings. Should I change them all in these files?